Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -1008,7 +1041,10 @@ def network_checkpoint(self):
                 if self.pstate_for_restart is not None:
                     resdata = self.pstate_for_restart.get('state', 'engineresumedata')
             elif isinstance(self.tdef, TorrentDef):
-                resdata = self.handle.save_resume_data()
+                self.save_resume_data()
+
+                # whether it's already called or not
+                resdata = self.resume_data
```

Comment:
calling save_resume_data() doesn't make self.resume_data contain the data right away. Also,  the pstate dict is being updated on the alert handler, so what follows doesn't make much sense anymore?